packages/base/src/widget.ts

```diff
@@ -991,67 +1000,61 @@ export class DOMWidgetView extends WidgetView {
     this.updateTooltip();
   }
 
-  setLayout(layout: LayoutModel, oldLayout?: LayoutModel): void {
+  async setLayout(layout: LayoutModel, oldLayout?: LayoutModel): Promise<void> {
```

Comment:
As earlier, I'm not ssure this old public method should be converted to `async`

Reply:
I see your point here. Especially since the outer promises in this method (layoutPromise and possibly create_child_view) are used to chain promises into a queue to be assigned to layoutPromise. Seems like this method should remain sync, however the inner promise `displayed` could be awaited. 

Therefore only the resolve of the `create_child_view` needs to be async.